azure/defaults_test.go

```diff
@@ -24,11 +24,109 @@ import (
 	"sync"
 	"testing"
 
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 	"github.com/Azure/go-autorest/autorest"
 	. "github.com/onsi/gomega"
 	"sigs.k8s.io/cluster-api-provider-azure/util/tele"
 )
 
+// TestARMClientOptions tests the `ARMClientOptions()` factory function.
+func TestARMClientOptions(t *testing.T) {
+	g := NewWithT(t)
+
+	tests := []struct {
+		name          string
+		cloudName     string
+		expectedCloud cloud.Configuration
+		expectError   bool
+	}{
+		{
+			name:          "should return default client options if cloudName is empty",
+			cloudName:     "",
+			expectedCloud: cloud.Configuration{},
+		},
+		{
+			name:          "should return Azure public cloud client options",
+			cloudName:     PublicCloudName,
+			expectedCloud: cloud.AzurePublic,
+		},
+		{
+			name:          "should return Azure China cloud client options",
+			cloudName:     ChinaCloudName,
+			expectedCloud: cloud.AzureChina,
+		},
+		{
+			name:          "should return Azure government cloud client options",
+			cloudName:     USGovernmentCloudName,
+			expectedCloud: cloud.AzureGovernment,
+		},
+		{
+			name:        "should return error if cloudName is unrecognized",
+			cloudName:   "AzureUnrecognizedCloud",
+			expectError: true,
+		},
+	}
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
I'd vote to move the `NewWithT` here:
```suggestion
			t.Parallel()
			g := NewWithT(t)
```